lib/ansible/modules/network/meraki/meraki_static_route.py

```diff
@@ -0,0 +1,386 @@
+
+options:
+    auth_key:
+        description:
+        - Authentication key provided by the dashboard.
+        - Required if environmental variable MERAKI_KEY is not set.
+        type: str
+    state:
+        description:
+        - Create or modify an organization.
+        choices: [ absent, query, present ]
+        default: present
+        type: str
+    net_name:
+        description:
+        - Name of a network.
+        type: str
+    net_id:
+        description:
+        - ID number of a network.
+        type: str
+    org_name:
+        description:
+        - Name of organization associated to a network.
+        type: str
+    org_id:
+        description:
+        - ID of organization associated to a network.
+        type: str
+    name:
+        description:
+        - Descriptive name of the static route.
+        type: str
+    subnet:
+        description:
+        - CIDR notation based subnet for static route.
+        type: str
+    gateway_ip:
+        description:
+        - IP address of the gateway for the subnet.
+        type: str
+    route_id:
+        description:
+        - Unique ID of static route.
+        type: str
+    fixed_ip_assignments:
+        description:
+        - List of fixed MAC to IP bindings for DHCP.
+        type: list
+        suboptions:
+            mac:
+                description:
+                - MAC address of endpoint.
+                type: str
+            ip:
+                description:
+                - IP address of endpoint.
+                type: str
+            name:
+                description:
+                - Hostname of endpoint.
+                type: str
+    reserved_ip_ranges:
+        description:
+        - List of IP ranges reserved for static IP assignments.
+        type: list
+        suboptions:
+            start:
+                description:
+                - First IP address of reserved range.
+                type: str
+            end:
+                description:
+                - Last IP address of reserved range.
+                type: str
+            comment:
+                description:
+                - Human readable description of reservation range.
+                type: str
+    enabled:
+        description:
+        - Indicates whether static route is enabled within a network.
+        type: bool
+
+
+author:
+    - Kevin Breit (@kbreit)
+extends_documentation_fragment: meraki
+'''
+
+EXAMPLES = r'''
+- name: Create static_route
+  meraki_static_route:
+    auth_key: abc123
+    state: present
+    org_name: YourOrg
+    net_name: YourNet
+    name: Test Route
+    subnet: 192.0.1.0/24
+    gateway_ip: 192.168.128.1
+  delegate_to: localhost
+
+- name: Update static route with fixed IP assignment
+  meraki_static_route:
+    auth_key: abc123
+    state: present
+    org_name: YourOrg
+    net_name: YourNet
+    route_id: d6fa4821-1234-4dfa-af6b-ae8b16c20c39
+    fixed_ip_assignments:
+      - aa:bb:cc:dd:ee:ff:
```

Comment:
I think this is weird and will confuse people. Why not simply go for a list of dictionaries, which is a lot easier than a list of dictionaries with dictionaries.
```yaml
    fixed_ip_assignments:
    - mac: aa:bb:cc:dd:ee:ff
      ip: 192.0.1.11
      comment: Server
```

Looking at the argspec, this is how it ought to be, even.

Reply:
Fixed.